### PR TITLE
Tour - Preview Mode fixes

### PR DESCRIPTION
**Pull request by roperzh**
OK to merge.

Medium review.

Asana task link: https://app.asana.com/0/480796620059175/537861811113865

Changes in this PR:

- [x] Refactor preview mode to control all the state management in <Creator /> and pass that state as a prop to children components instead of having logic all over the place.
- [x] Disable preview mode when the user leaves the "Congratulations" step
- [x] Disable the tour spotlight on the "Congratulations" step
- [x] Allow preview mode to be disabled by clicking on the library or the timeline

Notes for the reviewer:

Left some notes as a comments, please check that.

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [x] Did the [E2E checklist](https://haiku.quip.com/dGqeAEVSWdmg)
- [x] Expanded the [QA checklist](https://haiku.quip.com/rqwsAPsCGxqT)
- [x] Ran the automated tests and linted the code
- [ ] Wrote an automated test covering new functionality

### CODE REVIEW

File: packages/haiku-serialization/src/bll/Project.js [hunk 1]
```diff
@@ -153,10 +153,9 @@ class Project extends BaseModel {
     const fileOptions = this.getFileOptions()
     return (
       fileOptions &&
-      fileOptions.methodsToShortCircuit &&
+      fileOptions.whitelistedMethods &&
       (
-        fileOptions.methodsToShortCircuit[method] ||
-        fileOptions.methodsToShortCircuit['*']
+        !fileOptions.whitelistedMethods[method]
```
Comment (roperzh) [inline packages/haiku-serialization/src/bll/Project.js:158]: @matthewtoast how do you feel about this? I need to ping creator from the timeline in order to disable preview mode if the timeline overlay is clicked.

Comment (tsortwehttam) [reply packages/haiku-serialization/src/bll/Project.js:158]: Note: Once we get past launch, this interactionMode stuff might be a good piece to slice off and move into Envoy. 

For now, this logic has about the same level of hackiness as my earlier code, so it's probably fine. Going to review in context of the rest of this.

File: packages/haiku-creator/src/react/Creator.js [hunk 2]
```diff
@@ -712,6 +735,13 @@ export default class Creator extends React.Component {
               }
             })
 
+            projectModel.on('remote-update', (what) => {
+              switch (what) {
+                case 'setInteractionMode':
+                  return this.disablePreviewMode()
```
Comment (tsortwehttam) [inline packages/haiku-creator/src/react/Creator.js]: The method received is `'setInteractionMode'` but the logic assumes that any call here wants to `disablePreviewMode`. Could this cause a problem?

Comment (roperzh) [reply packages/haiku-creator/src/react/Creator.js]: This is intended, but is assuming that everytime we set preview mode from a remote source we want to disable it. I will update the logic to be more clear.

Comment (roperzh) [reply packages/haiku-creator/src/react/Creator.js]: Fix pushed :) thanks for noticing this

**Comment by tsortwehttam** [comment]
LGTM, merging.

**Pull request merged**